### DISCUSSION TITLE
how to customise the I2C speed

### DISCUSSION THREAD

**Discussion by danny1209** [OE4T/tegra-demo-distro#211] [Q&A] [answered]
Hi all, 

**tegra-vii2c 546c0000.i2c: no acknowledge from address 0x10**
my device with the address 0x10 is connected to this I2C. It happens not by every reading and writing, only randomly, seems the i2c not stable. I tried to find the reason, somebody said, it´s caused by too high clock frequency of i2c. therefore I want to know which frequency it uses for this i2c and how can I customise it? thanks in advance!

**Comment by ichergui** [comment, marked as answer]
Hey @danny1209 
Are you using the Nvidia devkit or custom HW (custom carrier board) ?
Maybe you have to adjust the clock frequency for that specific device (check the datasheet).

**Reply by danny1209** [reply]
Hi @ichergui 
yes, thanks! the problem was solved by setting a lower frequency for the i2c. in the file tegra210-porg-p3448-common.dtsi 
![image](https://user-images.githubusercontent.com/46848895/176670957-99244d8e-9da0-48ad-824b-9d331ffe9c36.png)

**Reply by danny1209** [reply]
@ichergui I must say the issue is still noch there...
